### PR TITLE
Remove unnecessary HTML elements, such as html, head, body from HTML, generated via RST.

### PR DESCRIPTION
**Pull request by svetlyak40wt**

### CODE REVIEW

File: plugins/rst.lisp [hunk 1]
```diff
@@ -4,20 +4,37 @@
...
   (:export #:enable))
 
 (in-package :coleslaw-rst)
 
+
+(defmethod visit-node :after ((writer html-writer) (document document))
+  "This method removes unnecessary HTML elements, such as html, head, body
+and make docutils output only html fragment with document itself."
+  
+  (setf (slot-value writer 'parts) '(body-pre-docinfo
```
Comment (PuercoPop) [inline plugins/rst.lisp]: Instead of adding an after method wouldn't passing a `:parts` initarg `'(body-pre-doc-info body)` to the writer in `RENDER-TEXT` accomplish our goal? ie

```lisp
(let ((writer (make-instance 'html-writer :parts '(body-predocoinfo body)))
          ..)
      ...)
```

File: plugins/rst.lisp [hunk 2]
```diff
@@ -4,20 +4,37 @@
...
+                                     body)))
+
+
 (defmethod render-text (text (format (eql :rst)))
   (register-settings-spec '((:generator nil)
                             (:datestamp nil)))
   (with-output-to-string (str)
     (let ((writer (make-instance 'html-writer))
           (document (read-rst text)))
-      (visit-node writer document)
```
Comment (PuercoPop) [inline plugins/rst.lisp]: Why is this line no longer necessary? The reason should be in the commit message

File: plugins/rst.lisp [hunk 3]
```diff
@@ -4,20 +4,37 @@
...
+                 )
   (:export #:enable))
 
 (in-package :coleslaw-rst)
 
+
+(defmethod visit-node :after ((writer html-writer) (document document))
+  "This method removes unnecessary HTML elements, such as html, head, body
+and make docutils output only html fragment with document itself."
+
```
Comment (PuercoPop) [inline plugins/rst.lisp]: This newline has a whitespace that should be removed (`M-\` in Emacs)

File: plugins/rst.lisp [hunk 4]
```diff
@@ -4,20 +4,37 @@
...
+                #:read-rst
+                #:register-settings-spec
+                #:visit-node
+                #:write-document)
+  (:import-from #:docutils.writer.html
+                #:html-writer
+                #:parts
+                #:body-pre-docinfo
+                #:body
+                 )
```
Comment (PuercoPop) [inline plugins/rst.lisp]: The paren shouldn't be hanging but at the end of `#:body`

File: plugins/rst.lisp (hunk 1 above)
Comment (PuercoPop) [inline plugins/rst.lisp]: To answer my own question, it appears that docutils then signals unbound slot on head among other things

Comment (Ferada) [inline plugins/rst.lisp]: Could still just be changed to mention all of the parts, but not set any values for `head-prefix`, `body-prefix` and `body-suffix`. It's a bit more declarative than adding a new method, but then again, I don't think that matters too much.

File: plugins/rst.lisp (hunk 2 above)
Comment (PuercoPop) [inline plugins/rst.lisp]: @Ferada Thanks for looking into this. So then this should be part of a separate commit, right?

File: plugins/rst.lisp (hunk 1 above)
Comment (PuercoPop) [inline plugins/rst.lisp]: The main reason I worry about defining a new method is that it affects potentially other users of docutils in the same image. Although this is highly unlikely

File: plugins/rst.lisp (hunk 2 above)
Comment (Ferada) [inline plugins/rst.lisp]: Arguably yes.

File: plugins/rst.lisp (hunk 4 above)
Comment (svetlyak40wt) [inline plugins/rst.lisp]: Ok, fixed.

File: plugins/rst.lisp (hunk 2 above)
Comment (svetlyak40wt) [inline plugins/rst.lisp]: @Ferada you was right. This works. Fixed.

**Comment by PuercoPop** [comment]
@svetlyak40wt Thanks, I'll split and merge it later today.

**Comment by PuercoPop** [comment]
Merged on https://github.com/kingcons/coleslaw/commit/808bdb50c309bfb6a44cb5f29b59b08800562d08

**Comment by kingcons** [comment]
\o/
